db/table_cache.cc

```diff
@@ -247,13 +247,15 @@ InternalIterator* TableCache::NewIterator(
     }
   }
   if (s.ok() && range_del_agg != nullptr && !options.ignore_range_deletions) {
-    std::unique_ptr<InternalIterator> range_del_iter(
-        table_reader->NewRangeTombstoneIterator(options));
-    if (range_del_iter != nullptr) {
-      s = range_del_iter->status();
-    }
-    if (s.ok()) {
-      s = range_del_agg->AddTombstones(std::move(range_del_iter));
+    if (range_del_agg->AddFile(fd.GetNumber())) {
```

Comment:
We should only call `AddFile` if `RangeDelAggregator::rep_` is non-nullptr. That way we avoid `std::set` operations for users who aren't using range deletions. Considering that, we probably should make the set a member variable on `Rep`.